### PR TITLE
Add release notes for BUcash 1.4.0.0

### PR DESCRIPTION
**Pull request by sickpig**
I'm going to add one more commit to include the same file also in `doc/release-notes/` folder and submit a new PR for `dev` once the review process will end.

### CODE REVIEW

**Comment by awemany** [comment]
Maybe add a comment that both `blocksdb` and `graphene` are experimental, and also add a comment that both are switch off by default?

Otherwise, looks good!

**Comment by sickpig** [comment]
fair points. will do. 
edit: done

File: doc/release-notes-bucash.md [hunk 1]
```diff
@@ -36,68 +37,567 @@ In case you decide to downgrade from BUcash 1.2.0.1, or greater, to a version ol
...
 
-- provide test infrastructure to create transactions with more complex script
-- various fix and improvements applied to BU request manager
-- fix socket select bug on windows platform
-- fix a problem with incoming connections due to node IP address discoverability
-- Ensure that EB and datacarrier size are set properly after a node restart
-- fix spurious disconnect due to block download crowding out getheaders
-- Prune slow peers during initial sync
+- Graphene Relay: A protocol for efficiently relaying blocks across a blockchain's network (experimental)
+- blocksdb: Add leveldb as an alternative storage method for blocks and undo data
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: add "(experimental, on-disk blocksdb data formats may change in subsequent releases, requiring resync)"

File: doc/release-notes-bucash.md [hunk 2]
```diff
@@ -36,68 +37,567 @@ In case you decide to downgrade from BUcash 1.2.0.1, or greater, to a version ol
...
+- Double Spend Relaying
+- BIP 135: Generalized version bits voting
+- clean up shadowing/thread clang warn
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: add "ings"
clean up shadowing/thread clang warnings

File: doc/release-notes-bucash.md (hunk 1 above)
Comment (sickpig) [reply doc/release-notes-bucash.md]: done

File: doc/release-notes-bucash.md [hunk 3]
```diff
@@ -36,68 +37,567 @@ In case you decide to downgrade from BUcash 1.2.0.1, or greater, to a version ol
...
+- Update depends libraries
+- Rework of the Bitcoin fuzzer command line driver tool
+- Add stand alone cpu miner to the set of binaries
+- Cashlib: create a shared library to make creating wallets easier
+- Imporva QA machinery (travis mainly)
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: "Imporva"?  maybe "Improve"

File: doc/release-notes-bucash.md [hunk 4]
```diff
@@ -36,68 +37,567 @@ In case you decide to downgrade from BUcash 1.2.0.1, or greater, to a version ol
...
-- Prune slow peers during initial sync
+- Graphene Relay: A protocol for efficiently relaying blocks across a blockchain's network (experimental)
+- blocksdb: Add leveldb as an alternative storage method for blocks and undo data
+- Double Spend Relaying
+- BIP 135: Generalized version bits voting
+- clean up shadowing/thread clang warn
+- Update depends libraries
+- Rework of the Bitcoin fuzzer command line driver tool
+- Add stand alone cpu miner to the set of binaries
+- Cashlib: create a shared library to make creating wallets easier
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: add something like: "(experimental) This library factors useful functionality out of bitcoind into a separate shared library that is callable from higher level languages. Currently supports transaction signing, additional functionality TBD."

File: doc/release-notes-bucash.md (hunk 3 above)
Comment (sickpig) [reply doc/release-notes-bucash.md]: typo, fixed

File: doc/release-notes-bucash.md [hunk 5]
```diff
@@ -36,68 +37,567 @@ In case you decide to downgrade from BUcash 1.2.0.1, or greater, to a version ol
...
-- fix spurious disconnect due to block download crowding out getheaders
-- Prune slow peers during initial sync
+- Graphene Relay: A protocol for efficiently relaying blocks across a blockchain's network (experimental)
+- blocksdb: Add leveldb as an alternative storage method for blocks and undo data
+- Double Spend Relaying
+- BIP 135: Generalized version bits voting
+- clean up shadowing/thread clang warn
+- Update depends libraries
+- Rework of the Bitcoin fuzzer command line driver tool
+- Add stand alone cpu miner to the set of binaries
```
Comment (gandrewstone) [inline doc/release-notes-bucash.md]: add "This cpu miner showcases the new mining RPC calls, provides a template for development of mining pool software, and is valuable for regtest/testnet mining.  CPU mining on mainnet with this or any other software is an exercise in futility"

File: doc/release-notes-bucash.md (hunk 4 above)
Comment (sickpig) [reply doc/release-notes-bucash.md]: done

File: doc/release-notes-bucash.md (hunk 5 above)
Comment (sickpig) [reply doc/release-notes-bucash.md]: done

**Pull request merged**